### PR TITLE
Store streams with index ranges

### PR DESCRIPTION
**Pull request by kroepke**
Connect #3128

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/indexer/indices/Indices.java [hunk 1]
```diff
@@ -589,19 +603,30 @@ public TimestampStats timestampStatsOfIndex(String index) {
...
         final Min minAgg = f.getAggregations().get("ts_min");
         final DateTime min = new DateTime((long) minAgg.getValue(), DateTimeZone.UTC);
         final Max maxAgg = f.getAggregations().get("ts_max");
         final DateTime max = new DateTime((long) maxAgg.getValue(), DateTimeZone.UTC);
+        List<String> streamIds = null;
+        final Terms streams = f.getAggregations().get("streams");
+        if (!streams.getBuckets().isEmpty()) {
+            streamIds = ImmutableList.copyOf(streams.getBuckets().stream()
+                    .map(Terms.Bucket::getKeyAsString)
+                    .collect(toSet()));
```
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/indexer/indices/Indices.java]: You use `toSet()` but assign it to a `List<String>`. How about switching `IndexRangeStats` over to `Set<String>` for the steams field?

Comment (kroepke) [inline graylog2-server/src/main/java/org/graylog2/indexer/indices/Indices.java]: Mongojack didn't work with Set, so I copy it into a list for storing.
It's only important that the list does not contain duplicates, which is achieved by the intermediate set.

**Comment by garybot2** [comment]
Looks like we would not be able to merge this PR because of merge conflicts. Please rebase, fix conflicts, and force push to your branch.

File: graylog2-server/src/main/java/org/graylog2/migrations/V20161130141500_DefaultStreamRecalcIndexRanges.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        final IndexSet defaultIndexSet = optDefaultIndexSet.get();
+
+        if (!cluster.isConnected()) {
+            LOG.info("Cluster not connected yet, delaying migration until it is reachable.");
+            while (true) {
+                try {
+                    cluster.waitForConnectedAndDeflectorHealthy();
+                    break;
+                } catch (InterruptedException | TimeoutException e) {
+                    e.printStackTrace();
```
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/migrations/V20161130141500_DefaultStreamRecalcIndexRanges.java]: Please log a proper error here. And do we need to do anything when that happens or are we just continue to the next loop iteration?

Comment (kroepke) [inline graylog2-server/src/main/java/org/graylog2/migrations/V20161130141500_DefaultStreamRecalcIndexRanges.java]: this is left over from automatic "try-catch adding" and should be a warning, I think. timeouts and spurious interrupts can happen. I'll fix.

**Pull request merged**